src/util.c

```diff
@@ -1254,12 +1315,28 @@ strtok_r(char *str, const char *delim, char **saveptr)
 }
 #endif
 
+extern char *current_working_dir;
+
 // Create an empty temporary file. *fname will be reallocated and set to the
 // resulting filename. Returns an open file descriptor to the file.
 int
 create_tmp_fd(char **fname)
 {
-	char *template = format("%s.%s", *fname, tmp_string());
+	char* template = NULL;
+#ifdef _WIN32
+		int maxlen=190;
+		int pathlen = strlen(format("%s.%s", *fname, tmp_string()));
+
+		if(pathlen > maxlen && atof(format("%ld.%ld", (DWORD)(LOBYTE(LOWORD(GetVersion()))), (DWORD)(HIBYTE(LOWORD(GetVersion()))))) > 6.1) {
```

Comment:
Can't this be something like GetVersion() > 0x0601? (I have not looked up the exact semantics, just guessing). You extra LOBYTE and HIBYTE and then concatenate them again wir format and then parse them via atof. Sounds like converting back and forth.